NitroxClient/GameLogic/RemotePlayer.cs

```diff
@@ -283,6 +287,45 @@ private void UpdateEquipmentVisibility()
             playerModelManager.UpdateEquipmentVisibility(new ReadOnlyCollection<TechType>(equipment.ToList()));
         }
         
+        /// <summary>
+        /// Makes the RemotePlayer recognizable as an obstacle for buildings.
+        /// </summary>
+        private void SetupBody()
+        {
+            RemotePlayerIdentifier identifier = Body.AddComponent<RemotePlayerIdentifier>();
+            identifier.RemotePlayer = this;
+            
+            if (Player.mainCollider is CapsuleCollider refCollider)
+            {
+                // This layer lets us have a collider as a trigger without preventing its detection as an obstacle
+                Body.layer = LayerID.Useable;
+                Collider = Body.AddComponent<CapsuleCollider>();
+                
+                Collider.center = Vector3.zero;
+                Collider.radius = refCollider.radius;
+                Collider.direction = refCollider.direction;
+                Collider.contactOffset = refCollider.contactOffset;
+                Collider.isTrigger = true;
+            }
+        }
+
+        /// <summary>
+        /// Change two parameters of the collider depending on the state of the player
+        /// </summary>
+        private void UpdateCollider(AnimChangeType type, AnimChangeState state)
+        {
+            if (type == AnimChangeType.UNDERWATER && state == AnimChangeState.ON)
```

Comment:
That way if the type is not UNDERWATER we always jump into the else, is this intended?

Reply:
yes, I'll modify the comment above to explain that there are two collisions behaviours depending on whether or not the player is underwater